### ISSUE TITLE
Should we warn about `SET DEFAULT` w/ update inside a transaction?

### ISSUE THREAD

**Issue by sbdchd** [sbdchd/squawk#115]
Here's a migration generated by Django that doesn't generate a warning from Squawk

```sql
BEGIN;
--
-- Alter field col_name on foo_tabl
--
ALTER TABLE "foo_tbl" ALTER COLUMN "col_name" SET DEFAULT false;
UPDATE "foo_tbl" SET "col_name" = false WHERE "col_name" IS NULL;
ALTER TABLE "foo_tbl" ALTER COLUMN "col_name" SET NOT NULL;
ALTER TABLE "foo_tbl" ALTER COLUMN "col_name" DROP DEFAULT;
COMMIT;
```

code change in python land was:

```python
class Foo(Model):
    field_name = models.BooleanField(null=True)
```

being replaced by:

```python
class Foo(Model):
    field_name = models.BooleanField(null=False, default=False)
```

**Comment by chdsbd** [comment]
`SET DEFAULT false` blocks reads to `foo_tbl`.

`UPDATE` could take a while to run and hold open the transaction.

**Comment by chdsbd** [comment]
I think this is the proper alternative
```sql
ALTER TABLE "foo_tbl" ALTER COLUMN "col_name" SET DEFAULT false;

-- prevent new null rows
ALTER TABLE "foo_tbl" ADD CONSTRAINT col_name_null
    CHECK ("col_name" IS NOT NULL) NOT VALID;

-- backfill column so it's not null
UPDATE "foo_tbl" SET "col_name" = false WHERE "col_name" IS NULL;
-- verify existing columns are not-null
ALTER TABLE "foo_tbl" VALIDATE CONSTRAINT col_name_null;
-- remove default
ALTER TABLE "foo_tbl" ALTER COLUMN "col_name" DROP DEFAULT;
```

**Comment by chdsbd** [comment]
Closing in favor of #123.
